CMakeLists.txt

```diff
@@ -551,31 +551,9 @@ if(ZEPHYR_GIT_INDEX)
   set(git_dependency ${ZEPHYR_GIT_INDEX})
 endif()
 
-add_custom_command(
-  OUTPUT ${PROJECT_BINARY_DIR}/include/generated/version.h
-  COMMAND ${CMAKE_COMMAND} -DZEPHYR_BASE=${ZEPHYR_BASE}
-    -DOUT_FILE=${PROJECT_BINARY_DIR}/include/generated/version.h
-    -DVERSION_TYPE=KERNEL
-    -DVERSION_FILE=${ZEPHYR_BASE}/VERSION
-    -DKERNEL_VERSION_CUSTOMIZATION="${KERNEL_VERSION_CUSTOMIZATION}"
-    ${build_version_argument}
-    -P ${ZEPHYR_BASE}/cmake/gen_version_h.cmake
-  DEPENDS ${ZEPHYR_BASE}/VERSION ${git_dependency}
-)
 add_custom_target(version_h DEPENDS ${PROJECT_BINARY_DIR}/include/generated/version.h)
 
 if(EXISTS ${APPLICATION_SOURCE_DIR}/VERSION)
-  add_custom_command(
-    OUTPUT ${PROJECT_BINARY_DIR}/include/generated/app_version.h
-    COMMAND ${CMAKE_COMMAND} -DZEPHYR_BASE=${ZEPHYR_BASE}
-      -DOUT_FILE=${PROJECT_BINARY_DIR}/include/generated/app_version.h
-      -DVERSION_TYPE=APP
-      -DVERSION_FILE=${APPLICATION_SOURCE_DIR}/VERSION
-      -DAPP_VERSION_CUSTOMIZATION="${APP_VERSION_CUSTOMIZATION}"
-      ${build_version_argument}
-      -P ${ZEPHYR_BASE}/cmake/gen_version_h.cmake
-    DEPENDS ${APPLICATION_SOURCE_DIR}/VERSION ${git_dependency}
-  )
   add_custom_target(app_version_h DEPENDS ${PROJECT_BINARY_DIR}/include/generated/app_version.h)
   add_dependencies(zephyr_interface app_version_h)
 endif()
```

Comment:
I suppose another option is to change `APP_VERSION_CUSTOMIZATION` to be a target property of `app_version_h`, then use a genex in place:
```cmake
add_custom_command(
  COMMAND ...
    -DVERSION_CUSTOMIZATION=$<TARGET_PROPERTY:app_version_h,VERSION_CUSTOMIZATION>
  ...
  VERBATIM
)
```

Not blocking this PR, but it seems slightly cleaner than having to duplicate the if-condition below. I'm curious about your thoughts.

Reply:
actually, that's a good idea.

Reply:
done.

Thanks for the idea.